api/jobserver/clear_task_cache.go

```diff
@@ -0,0 +1,76 @@
+package jobserver
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"code.cloudfoundry.org/lager"
+	"github.com/concourse/atc"
+	"github.com/concourse/atc/db"
+)
+
+func (s *Server) ClearTaskCache(pipeline db.Pipeline) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		logger := s.logger.Session("clear-task-cache")
+		jobName := r.FormValue(":job_name")
+		stepName := r.FormValue(":step_name")
+		cachePath := r.FormValue(atc.ClearTaskCacheQueryPath)
+
+		job, found, err := pipeline.Job(jobName)
+		if err != nil {
+			logger.Error("failed-to-get-job", err)
+			w.WriteHeader(http.StatusInternalServerError)
+			return
+		}
+
+		if !found {
+			logger.Debug("could-not-find-job", lager.Data{
+				"jobName":   jobName,
+				"stepName":  stepName,
+				"cachePath": cachePath,
+			})
+			w.WriteHeader(http.StatusNotFound)
+			return
+		}
+
+		rowsDeleted, err := job.ClearTaskCache(stepName, cachePath)
+
+		if rowsDeleted == 0 {
+			logger.Debug("could-not-find-cache-path", lager.Data{
+				"jobName":   jobName,
+				"stepName":  stepName,
+				"cachePath": cachePath,
+			})
+			w.WriteHeader(http.StatusNotFound)
```

Comment:
Hmm, what do you think about having this not be an error case and instead just return the number of caches that were deleted? Clearing caches is something that could easily be idempotent, but I can see why you'd want to surface *something* as the user may have typo'd the path or step name or something. Showing how many were deleted would let the user notice if they made a typo (because they'd expect a nonzero amount) or reason that maybe the cache actually wasn't there themselves. (It'd also be a nice confirmation to see how many were cleared.)

Reply:
Ah yes, I was a bit uncertain about returning an error here.  I think it's a great idea to just return the number of deleted caches - it nicely handles both cases where either the caches were successfully deleted or the user may have made a typo.  Will update the code with these changes.